code/ai/aicode.cpp

```diff
@@ -6159,6 +6168,57 @@ int check_ok_to_fire(int objnum, int target_objnum, weapon_info *wip)
 	return 1;
 }
 
+//	--------------------------------------------------------------------------
+//  Returns true if *aip has a line of sight to its current target.
+//	This is a computationally expensive operation, so use sparingly
+int check_los(int objnum, int target_objnum) {
+	vec3d& end = Objects[target_objnum].pos;
+	vec3d& start = Objects[objnum].pos;
+
+	for (int i = 0; i < Highest_object_index; i++) {
```

Comment:
Iterating through objects is generally done through ´obj_used_list´ (searching for that will 1quickly bring up how its used in this way). I don't know if its better that way, but its the more commonly used method.

Reply:
Goober also suggested this, so okay. I changed it